corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -691,13 +691,21 @@ hqDefine("cloudcare/js/form_entry/form_ui", [
                     element.serverError(null);
                 }
 
-                if (allChildren) {
-                    for (let i = 0; i < allChildren.length; i++) {
-                        if (allChildren[i].control >= constants.CONTROL_IMAGE_CHOOSE) {
-                            allChildren[i].filename = element.answer();
+                let findChildAndSetFilename = function (children) {
+                    for (var i = 0; i < children.length; i++) {
+                        var child = children[i];
+                        if (child.children && child.children.length > 0) {
+                            findChildAndSetFilename(child.children);
+                        } else if (child.control >= constants.CONTROL_IMAGE_CHOOSE) {
```

Comment:
for future proofing, should we check for the exact control value instead of doing `>=`? In case any control is added in the future

Reply:
this is a good idea since Martin was a bit confused about what exactly this did - put them all into an array to check against so it's explicit what we're checking for